rpcserver.go

```diff
@@ -1911,6 +1911,35 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 
 	localFundingAmt := btcutil.Amount(in.LocalFundingAmount)
 	remoteInitialBalance := btcutil.Amount(in.PushSat)
+
+	// If the FundMax flag is set, ensure that the acceptable minimum local
+	// amount adheres to the amount to be pushed to the remote, and to
+	// current rules, while also respecting the settings for the maximum
+	// channel size.
+	var fundUpToMaxAmt btcutil.Amount
+	if in.FundMax {
+		// We assume the configured maximum channel size to be the upper
+		// bound of our "maxed" out funding attempt.
+		fundUpToMaxAmt = btcutil.Amount(r.cfg.MaxChanSize)
+
+		// Since the standard non-fundmax flow requires the local
+		// funding amount to be at least in the amount of the initial
+		// remote balance(push amount) we need to adjust the local
+		// funding amount accordingly. We initially assume the minimum
+		// allowed channel size as local as local funding amount.
+		localFundingAmt = funding.MinChanFundingSize
+
+		// Now if that local funding amount is less than the initial
+		// remote balance we simply increment the initial remote balance
+		// by 1 satoshi and take that as local funding amount in order
+		// to fullfil the criterion. Whether or not this so determined
+		// local funding amount is actually available is ascertained
+		// later on in the lnwallet's reservation workflow.
+		if remoteInitialBalance >= localFundingAmt {
+			localFundingAmt = remoteInitialBalance + 1
```

Comment:
Yes you are right, the `+1` is not needed to satisfy the `if remoteInitialBalance >= localFundingAmt` check a few lines below.
Since we are now using `minFundingAmt` to represent the minimum channel cap this check is omitted. Whether or not this determined minimum cap will be enough to cover a channel opening regarding initial remote balance and reserve etc. is checked further downstream in `reservation.go`:
https://github.com/lightningnetwork/lnd/blob/ef8cf73d4ce4b485d47ded5fa219992574c928c1/lnwallet/reservation.go#L268